simplemma/lemmatizer.py

```diff
@@ -185,31 +198,21 @@ def _return_lemma(
     # filters
     if token.isnumeric():
         return token
-    # dictionary search
-    candidate = _simple_search(token, datadict, initial=initial)
-    # simple rules
-    if candidate is None and lang is not None and lang in APPLY_RULES:
-        candidate = APPLY_RULES[lang](token, greedy)
-    # decomposition
-    if candidate is None:  # and greedy is True
-        candidate = _dehyphen(token, datadict, greedy)
-    else:
-        newcandidate = _dehyphen(candidate, datadict, greedy)
-        if newcandidate is not None:
-            candidate = newcandidate
+
+    candidate = (
```

Comment:
This looks so much nicer and cleaner!!

Could we wrap the `_affix_search` and the `_greedy_search` in blocks to also fit them here?

Reply:
Not really but I tried to simplify the rest of the function as well.